src/components/RandomMeme/RandomMeme.js

```diff
@@ -0,0 +1,115 @@
+import React, {Component} from "react";
+import {
+    ListItemText , Grid, Container, Box, ListItemAvatar, Avatar, ListItem
+
+} from "@material-ui/core";
+import StarIcon from '@material-ui/icons/Star';
+import StyledComponents from "./styles";
+
+class RandomMeme extends Component {
+    constructor() {
+        super();
+        this.state = {
+            randomImg: "",
+            allMemeImg: [],
+            alt: "",
+            width: "",
+            height: ""
+        };
+    }
+
+    random() {
+        let randomNumber = Math.floor(
+            Math.random() * this.state.allMemeImg.length
+        );
+        this.setState({
+            randomImg: this.state.allMemeImg[randomNumber].url,
+            alt: this.state.allMemeImg[randomNumber].name,
+            width: this.state.allMemeImg[randomNumber].width,
+            height: this.state.allMemeImg[randomNumber].height
+        });
+    }
+
+    imgSize(width, height) {
+        if (height > 1500 || width > 1500) {
+            return 0.2;
+        }
+        if (height > 1000 || width > 1000){
+            return 0.4;
+        }
+        if (height > 560 && height < 1000 || width > 560 && width < 1000) {
```

Comment:
Dorzuciłbym jakieś nawiasy dookoła warunków które sprawdzają się w pierwszej kolejności. Łatwiej się to czyta.